zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -81,6 +88,62 @@ use input_selection::{
     GreedyInputSelector, GreedyInputSelectorError, InputSelector, InputSelectorError,
 };
 
+/// Describes the key inputs and UFVK for an account that was derived from a ZIP-32 HD seed and account index.
+#[derive(Debug, Clone)]
+pub struct HdSeedAccount(
+    pub HdSeedFingerprint,
+    pub zip32::AccountId,
+    pub UnifiedFullViewingKey,
+);
+
+/// Represents an arbitrary account for which the seed and ZIP-32 account ID are not known
+/// and may not have been involved in creating this account.
+#[derive(Debug, Clone)]
+pub enum ImportedAccount {
+    /// An account that was imported via its full viewing key.
+    Full(UnifiedFullViewingKey),
+}
+
+/// Describes an account in terms of its UVK or ZIP-32 origins.
+#[derive(Debug, Clone)]
+pub enum Account {
+    /// Inputs for a ZIP-32 HD account.
+    Zip32(HdSeedAccount),
+    /// Inputs for an imported account.
+    Imported(ImportedAccount),
+}
+
+impl Account {
+    /// Returns the default Unified Address for the account,
+    /// along with the diversifier index that generated it.
+    ///
+    /// The diversifier index may be non-zero if the Unified Address includes a Sapling  
+    /// receiver, and there was no valid Sapling receiver at diversifier index zero.
+    pub fn default_address(
+        &self,
+        request: UnifiedAddressRequest,
+    ) -> Result<(UnifiedAddress, DiversifierIndex), AddressGenerationError> {
+        match self {
+            Account::Zip32(HdSeedAccount(_, _, ufvk)) => ufvk.default_address(request),
+            Account::Imported(ImportedAccount::Full(ufvk)) => ufvk.default_address(request),
+        }
+    }
+
+    /// Gets the unified full viewing key for this account, if available.
+    ///
+    /// Accounts initialized with an incoming viewing key will not have a unified full viewing key.
+    pub fn ufvk(&self) -> Option<&UnifiedFullViewingKey> {
+        match self {
+            Account::Zip32(HdSeedAccount(_, _, ufvk)) => Some(ufvk),
+            Account::Imported(ImportedAccount::Full(ufvk)) => Some(ufvk),
+        }
+    }
+
+    // TODO: When a UnifiedIncomingViewingKey is available, add a function that
+    // will return it. Even if the Account was initialized with a UFVK, we can always
+    // derive a UIVK from that.
```

Comment:
There are two UIVKs that could be returned from such a method: the external UIVK, and the internal UIVK. There are a few possibilities:
- This method takes a `Scope` argument and returns `Option<UIVK>` (in case the account is for an imported `(Scope, UIVK)` in which case we can't obtain the other scoped UIVK).
- We restrict ourselves to importing external UIVKs (and just hope the user doesn't lie to us when importing a UIVK, or if `Account` remains transparently constructable then hope the user doesn't lie to us when manually constructing the enum themselves), and then only return the external UIVK from here (and maybe have a separate method returning `Option` for the internal UIVK).

We also might end up having different types for external and internal UIVKs (or a common struct with a type marker), since we NEVER want to expose addresses for internal UIVKs, and the easiest way to enforce that is to simply not have those methods be available. (The protocol-specific crates don't currently do this, but we might also want to consider this there.)

Reply:
My take is that we will only ever use external UIVKs here.

Reply:
I've updated the comment to reflect this will always be external scope only.